### ISSUE TITLE
Support start scripts by RobotFramework LS project (in vscode)

### ISSUE THREAD

**Issue by Ka55i0peia** [robocorp/robotframework-lsp#280]
**Is your feature request related to a problem? Please describe.**

When a project uses  ["start scripts" like they have been introduced by robot man pages](https://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#creating-start-up-scripts) (especially the a [python start script](https://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#python-example)) it is hard to integrate rf LSP project into those robot framework setups.

In the start script we are defining, e.g. [tags](https://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#tag-patterns) to execute/exclude test cases (next to variables made available for runtime and custom listeners being attached to the robot run).

**Describe alternatives you've considered**

It might be possible to embed them at `.vscode/launch.json` (for using debugger functionality of LSP) and define variables in `.vscode/settings.json` (for running LSP as language server/linter) like discussed at #261 .

But defining tags explicit can be some kind of "brain fuck" and is _not targetting when they are defined in a hard coded way_. Since tags are a usefull feature to select tests to execute for a curtain software variant. Thus this tags are more runtime parameters then hard coded config parameters. (E.g. adding robot command argument `-e market_aORmarket_bNOTmarket_c` + some variable values beeing constructed by a question which market shall be tested)

**Describe the solution you'd like**

Making some injection [where LSP runs robot_cli](https://github.com/robocorp/robotframework-lsp/blob/d5a21c1ae600d5fbdbf2d492fb121a80d26391a3/robotframework-ls/src/robotframework_debug_adapter/run_robot__main__.py#L539) works like a charm. (Replacing this module by our start script having the same python interface and calling `robot.run_cli` internally by inriching the command line arguments). But this works for the "using debugger functionality of LSP" only. But this not work "for running LSP as language server/linter". (I haven't found the entry point for running robot command here.)

**Comment by fabioz** [comment]
@Ka55i0peia I'm not really sure I follow on how you'd like to have that working on the language server...

I must say that in your specific use-case, if you have a tool to make the runs, I think that making the tool generate the project structure (i.e.: the settings and the launch) would be the way to go (so, what the language server needs is a way for you to configure those through files, which it already has).

**Comment by Ka55i0peia** [comment]
@fabioz Thanks. Looks like I skipped the mail (didn't see you've ansered.)

> I think that making the tool generate the project structure (i.e.: the settings and the launch) would be the way to go 

Yes, this is how we worked around this. (Anyhow it is limited by https://github.com/robocorp/robotframework-lsp/issues/239 )

For linting with RF LSP, tags might not be required. (I don't see a problematic use case.) For running debug session with RF LSP, the cli arguments (like variable, tags and listeners) can be assigned to task.json.
